tabular/src/autogluon/tabular/utils.py

```diff
@@ -150,3 +156,222 @@ def infer_eval_metric(problem_type: str) -> Scorer:
+    predict_func_kwargs : dict, default {}
+        Keyword arguments to be appended to calls to `predict_func(X, **kwargs)`.
+    transform_func : Callable[..., pd.DataFrame], default None
+        Transformation function that takes the raw input and transforms it row-wise to the input expected by `predict_func`.
+        Common examples include `model.preprocess` and `feature_generator.transform`.
+        If None, then no transformation is done on the data prior to calling `predict_func`.
+        This is necessary to compute importance of original data features in `X` prior to their transformation assuming `predict_func` does not perform the transformation already.
+            Example: `transform_func` is necessary to compute the importance of a text feature prior to being transformed into ngrams by `transform_func` when `predict_func` expects ngram features as input.
+    transform_func_kwargs : dict, default {}
+        Keyword arguments to be appended to calls to `transform_func(X, **kwargs)`.
+    time_limit : float, default None
+        Time in seconds to limit the calculation of feature importance.
+        If None, feature importance will calculate without early stopping.
+        A minimum of 1 full shuffle set will always be evaluated. If a shuffle set evaluation takes longer than `time_limit`, the method will take the length of a shuffle set evaluation to return regardless of the `time_limit`.
+        If `num_shuffle_sets==1`, `time_limit` will be ignored.
+    silent : bool, default False
+        Whether to suppress logging output
+
+    Returns
+    -------
+    Pandas `pandas.DataFrame` of feature importance scores with 3 columns:
+        index: The feature name.
+        'importance': The feature importance score.
+        'stddev': The standard deviation of the feature importance score. If NaN, then not enough num_shuffle_sets were used to calculate a variance.
+        'z_score': The z-score of the feature importance score. Equivalent to 'importance' / 'stddev'.
+            A z-score of +4 or higher indicates that the feature is almost certainly useful and should be kept.
+            A z-score of +2 indicates that the feature has a 97.5% chance of improving model quality when present.
+            A z-score that is 0 or negative indicates that the feature can likely be dropped without negative impact to model quality.
+            A z-score of NaN indicates that the feature's stddev was NaN or that the model predictions were never impacted by the feature (importance 0 and stddev 0). This indicates that the feature can safely be dropped.
+            A z-score of +inf or -inf indicates that `subsample_size` and/or `num_shuffle_sets` were too small to calculate variance for the feature (non-zero importance with zero stddev).
+    """
+    if num_shuffle_sets is None:
+        if time_limit is not None:
+            num_shuffle_sets = 10
+        else:
+            num_shuffle_sets = 1
```

Comment:
Your example should be flipped I believe.

Additionally: `if time_limit` is different from `if time_limit is not None` because of time_limit == 0 case so I have simplified to:

```
    if num_shuffle_sets is None:
        num_shuffle_sets = 1 if time_limit is None else 10
```